### PR TITLE
Use std threads support

### LINKED ISSUE #45: Merge our thread code back into std?
Some context: https://github.com/Meziu/ctru-rs/issues/44

* Pros
  1. Keeps us up-to-date with std.
  2. Not duplicating code. We could define an extension trait for 3ds specific data like affinity.
  4. Other code can still use std threads (though we should generally avoid this since the 3ds CPU works differently).
* Cons
  1. This allows 3rd party code to use threads, which might break since the default core is not preemptive (and we only get one thread on the preemptive system core).
      * TODO: if it's not preemptive, what is priority used for? Research how the application core scheduler works.
  2. We can't directly link std to libctru. So we'd need to pass the extra data (ex. affinity) through pthread or something.

### PR DESCRIPTION
**Pull request by AzureMarker**
@Meziu @ian-h-chamberlain 

Closes #45 

Requires:
- [x] https://github.com/Meziu/rust-horizon/pull/10

Feels good to delete a bunch of code.

I tried using `cargo 3ds test` with these changes though and got some weird linker errors. It looks like libctru is included twice, but pthread-3ds isn't at all? I'll try debugging more but let me know if you see issues.
Edit: The test issues are fixed now.

### CODE REVIEW

**Comment by AzureMarker** [comment]
> Were you trying to `cargo 3ds test` without specifying a package or `--lib` ?

This is probably the reason.

File: ctru-rs/examples/thread-basic.rs [hunk 1]
```diff
@@ -14,19 +13,19 @@ fn main() {
     let gfx = Gfx::default();
     let _console = Console::init(gfx.top_screen.borrow_mut());
 
-    let prio = thread::current().priority();
+    let prio = ctru::thread::priority();
```
[the remark this answers is not in the document]
Comment (AzureMarker) [reply ctru-rs/examples/thread-basic.rs:16]: I left it in so we would only minimally modify std, and leave libctru-specific extensions here (like CPU affinity). But I could also move these to std.

**Comment by ian-h-chamberlain** [comment]
@AzureMarker make sure your deps are up-to-date – `pthread_sigmask` at least should be present in the latest `pthread-3ds` and the others I think are no longer needed in the latest `horizon-std` as of https://github.com/Meziu/rust-horizon/commit/ef21c1c010c91476eb50f0765e640b09ea8af998 

I recall seeing the same link errors I think until I did a `cargo update` and pulled latest std

File: ctru-rs/examples/thread-basic.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (AzureMarker) [reply ctru-rs/examples/thread-basic.rs:18]: I plan on moving these to `std::os::horizon::thread`, though that means I need to find a way to give std the CPU affinity value. I'll look into https://github.com/Meziu/rust-horizon/pull/10#discussion_r803698363

**Comment by AzureMarker** [comment]
@ian-h-chamberlain I tried the test command again after updating everything and it works! I also don't see any panic messages in the console (thanks to std threads). Some odd happens though if I run it a bunch of times. When exiting the 5th run, I get an ARM exception: prefetch abort (kernel panic). I haven't debugged this at all (just reproduced it), so it might be related to the std threads, or it's always been there, but that was weird.

**Review by ian-h-chamberlain — APPROVED** [review]

File: ctru-rs/examples/thread-basic.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (AzureMarker) [reply ctru-rs/examples/thread-basic.rs]: This has been fixed

**Pull request merged**